server/apis/v1/dexauth.go

```diff
@@ -0,0 +1,177 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/oauth2"
+)
+
+// DexObject is a struct that holds details for dex handlers
+type DexObject struct {
+	clientID     string
+	clientSecret string
+	redirectURI  string
+
+	verifier *oidc.IDTokenVerifier
+	provider *oidc.Provider
+
+	// Does the provider use "offline_access" scope to request a refresh token
+	// or does it use "access_type=offline" (e.g. Google)?
+	offlineAsScope bool
+
+	client     *http.Client
+	stateNonce string // stateNonce is the nonce variable
+}
+
+func (d *DexObject) oauth2Config(scopes []string) *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     d.clientID,
+		ClientSecret: d.clientSecret,
+		Endpoint:     d.provider.Endpoint(),
+		Scopes:       scopes,
+		RedirectURL:  d.redirectURI,
+	}
+}
+
+// NewDexObject returns a new DexObject.
+// TODO: refactor data structure and make configurable
+func NewDexObject(ctx context.Context) *DexObject {
+	clientID := "example-app"
+	issuerURL := "http://numaflow-dex-server:5556/dex"
+	provider, err := oidc.NewProvider(ctx, issuerURL)
+	if err != nil {
+		log.Fatalf("failed to query provider %q: %v", issuerURL, err)
+	}
+	verifier := provider.Verifier(&oidc.Config{ClientID: clientID})
+
+	return &DexObject{
+		clientID:       clientID,
+		clientSecret:   "ZXhhbXBsZS1hcHAtc2VjcmV0",
```

Comment:
where is this secret coming from?

Reply:
I copied and pasted from the Dex example app. It's hardcoded. I don't think we need this. Will clean up later as part of `make Dex configurable` TODO.